osu.Game/Graphics/Containers/OsuLinkFlowContainer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics.Sprites;
+using System;
+using System.Collections.Generic;
+
+namespace osu.Game.Graphics.Containers
+{
+    public class OsuLinkFlowContainer : OsuLinkFlowContainer<OsuSpriteLink>
+    {
+        public OsuLinkFlowContainer(Action<SpriteText> defaultCreationParameters = null)
+            : base(defaultCreationParameters)
+        {
+        }
+    }
+
+    public class OsuLinkFlowContainer<T> : OsuTextFlowContainer
+        where T : OsuSpriteLink, new()
+    {
+        public override bool HandleInput => true;
+
+        public OsuLinkFlowContainer(Action<SpriteText> defaultCreationParameters = null)
+            : base(defaultCreationParameters)
+        {
+        }
+
+        protected override SpriteText CreateSpriteText() => new T();
+
+        /// <summary>
+        /// The colour for text (links override this). Will only be used for new text elements.
+        /// </summary>
+        public ColourInfo TextColour = Color4.White;

```

Comment:
why is this required?

Reply:
It's for messages with a "special" sender ('with background' in code). At the moment, the whole `contentFlow` changes colour which is not what we want since if we do that, the colours will overlap since the links will still be blue. For example: a special sender with yellow colour will have green-ish links with the old way of doing this. So the Sprites should be assigned the color directly instead of the whole container.